applications/models.py

```diff
@@ -139,19 +194,102 @@ class ApplicationPeriod(models.Model):
         verbose_name=_("Reservation period end"),
     )
 
+    public_display_begin = models.DateTimeField(
+        verbose_name=_("Public display begin"),
+    )
+    public_display_end = models.DateTimeField(
+        verbose_name=_("Public display end"),
+    )
+
     purposes = models.ManyToManyField(
         Purpose,
         verbose_name=_("Purposes"),
-        related_name="application_periods",
+        related_name="application_rounds",
         blank=True,
     )
 
+    service_sector = models.ForeignKey(
+        "spaces.ServiceSector",
+        verbose_name=_("Service sector"),
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+    )
+
+    @property
+    def status(self):
+        return self.get_status().status
+
+    @status.setter
+    def status(self, status):
+        self.set_status(status)
+
+    @status.getter
+    def status(self):
+        return self.get_status().status
+
+    def set_status(self, status, user=None):
+        if status not in ApplicationRoundStatus.get_statuses():
+            raise ValidationError(_("Invalid status"))
+        ApplicationRoundStatus.objects.create(
+            application_round=self, status=status, user=user
+        )
+
+    def get_status(self):
+        return self.statuses.last()
+
     def __str__(self):
         return "{} ({} - {})".format(
             self.name, self.reservation_period_begin, self.reservation_period_end
         )
 
 
+class ApplicationRoundBasket(models.Model):
+    CUSTOMER_TYPE_BUSINESS = "business"
+    CUSTOMER_TYPE_NONPROFIT = "nonprofit"
+    CUSTOMER_TYPE_INDIVIDUAL = "individual"
+
+    CUSTOMER_TYPE_CHOICES = (
+        (CUSTOMER_TYPE_BUSINESS, _("Business")),
+        (CUSTOMER_TYPE_NONPROFIT, _("Nonprofit")),
+        (CUSTOMER_TYPE_INDIVIDUAL, _("Individual")),
+    )
+
+    name = models.CharField(
+        verbose_name=_("Name"),
+        max_length=255,
+    )
+    application_round = models.ForeignKey(
+        ApplicationRound,
+        verbose_name=_("Application round"),
+        on_delete=models.CASCADE,
+        related_name="application_round_baskets",
+    )
+
+    purpose = models.ForeignKey(
+        "reservation_units.Purpose",
+        verbose_name=_("Purpose"),
+        related_name="application_round_baskets",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+    must_be_main_purpose_of_applicant = models.BooleanField(
+        verbose_name=_("Must be main purpose of the applicant"), default=False
+    )
+    customer_type = ArrayField(
+        models.CharField(max_length=50, choices=CUSTOMER_TYPE_CHOICES),
+    )
+    age_groups = models.ManyToManyField("reservations.AgeGroup")
+    home_city = models.CharField(max_length=255)
+    allocation = models.PositiveSmallIntegerField(
```

Comment:
This could maybe have more descriptive name, don't know what's a great term, maybe something like allocation_target_percentage, or maybe just even allocation_percentage

Reply:
Changed to allocation_percentage